lib/bolt/pal.rb

```diff
@@ -312,8 +312,16 @@ def get_plan_info(plan_name)
 
       # If it's a Puppet language plan, use strings to extract data. The only
       # way to tell is to check which filename exists in the module.
-      plan_file = plan_name.split('::', 2)[1] || 'init'
-      pp_path = File.join(mod, 'plans', "#{plan_file}.pp")
+      #
+      # It's easier to make 2 here than try to keep plan_dir clean
+      pp_plan = plan_name.split('::')[1..-1]
```

Comment:
We could split and then `File.join` the segments, in which case we could just interpolate the filename along with the file extension where we need it. That would remove a lot of the duplication.